File: 633.cpp
struct Solution {
    inline bool judgeSquareSum(int c) {
        long a = 0, b = static_cast<int>(sqrt(c)), d;
        while (a <= b){
            d = a * a + b * b;
            if (d == c){
                return true;
            } else if (d > c){
                --b;
            } else {
                ++a;
            }
        }
        return false;
    }
};

File: 633.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a = 0
        b = int(c ** (1 / 2))
        while a <= b:
            d = a * a + b * b
            if d == c:
                return True
            elif d > c:
                b -= 1
            else:
                a += 1
        return False
